iceoryx_posh/test/moduletests/test_mepoo_shared_pointer.cpp

```diff
@@ -105,11 +104,11 @@ class SharedPointer_Test : public Test
 
     void SetUp() override
     {
-        iox::rp::BaseRelativePointer::registerPtr(memory, 4096);
+        iox::rp::UntypedRelativePointer::registerPtr(memory, 4096);
```

Comment:
This should also be a free function in say the rp namespace. The registration is not tied to a specific type (and should not be to work properly as we drop type information in shared memory at several points).

Reply:
I'm not sure here. Is it not conceptually a completely valid use case to register a `void*` and later searching for it and using it with a type `T`. Of course under the condition that the size is correct.

Reply:
@mossmaurice Yes this is the use case we currently have. We register raw memory and do not know beforehand what kind of types we will store there. Besides, one segment will contain multiple types in general.

This means the registration mechanism always uses untyped pointers.